core/state_transition.go

```diff
@@ -356,8 +356,16 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 		st.state.SetNonce(msg.From(), st.state.GetNonce(sender.Address())+1)
 		ret, st.gas, vmerr = st.evm.Call(sender, st.to(), st.data, st.gas, st.value)
 	}
+
+	// refund gas first in case refunded address is the coinbase.
+	// otherwise refunding after adding balance to coinbase might result into
+	// an overflow
```

Comment:
refunding gas could trigger an overflow in the case that a transaction execution results in increasing the balance of the caller, so should this function return an error?

Reply:
we set to maxuint256